src/js/components/search/filters/agency/AgencyList.jsx

```diff
@@ -0,0 +1,219 @@
+/**
+ * AgencyList.jsx
+ * Created by Emily Gullo 12/23/2016
+ **/
+
+import React, { PropTypes } from 'react';
+import _ from 'lodash';
+
+import Typeahead from 'components/sharedComponents/Typeahead';
+import TypeaheadWarning from 'components/sharedComponents/TypeaheadWarning';
+import SelectedAgencies from './SelectedAgencies';
+
+const propTypes = {
+    autocompleteAwardingAgencies: PropTypes.array,
+    autocompleteFundingAgencies: PropTypes.array,
+    onSelect: PropTypes.func,
+    customClass: PropTypes.string,
+    keyValue: PropTypes.string,
+    internalValue: PropTypes.string,
+    formatter: React.PropTypes.func,
+    tabIndex: PropTypes.number,
+    handleTextInput: PropTypes.func,
+    isRequired: PropTypes.bool,
+    agencyType: PropTypes.string,
+    checkValidity: PropTypes.func
+};
+
+const defaultProps = {
+    customClass: '',
+    formatter: null,
+    keyValue: 'agency',
+    internalValue: 'agency_name',
+    tabIndex: null,
+    isRequired: false,
+    errorHeader: '',
+    errorMessage: ''
+};
+
+export default class AgencyList extends Typeahead {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showWarning: false,
+            errorMessage: null,
+            errorHeader: null
+        };
+
+        this.timeout = null;
+    }
+
+    loadValues() {
+        const awardingValuesList = [];
+        const fundingValuesList = [];
+        if (this.props.autocompleteAwardingAgencies.length > 0) {
+            this.props.autocompleteAwardingAgencies.forEach((item) => {
+                awardingValuesList.push(item);
+                const key =
+                `<b>${item}</b>`;
+                this.dataDictionary[key] = item;
+            });
+        }
+        if (this.props.autocompleteFundingAgencies.length > 0) {
+            this.props.autocompleteFundingAgencies.forEach((item) => {
+                fundingValuesList.push(item);
+                const key =
+                `<b>${item}</b>`;
+                this.dataDictionary[key] = item;
+            });
+        }
+
+        if (this.props.agencyType === "Awarding") {
+            this.typeahead.list = awardingValuesList;
+        }
+        else {
+            this.typeahead.list = fundingValuesList;
+        }
+
+        this.typeahead.replace = () => {
+            this.typeahead.input.value = "";
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isEqual(prevProps.autocompleteAwardingAgencies,
```

Comment:
See comment in `AgencyListContainer`. This component shouldn't update on both prop changes; it should only be observing the agency data that applies to the current agency type.